### PR TITLE
make integration-tests work for cumulus-1043

### PR DESCRIPTION
**Pull request by scisco**

### CODE REVIEW

File: example/spec/parallel/ingestGranule/IngestGranuleSuccessSpec.js [hunk 1]
```diff
@@ -326,7 +326,9 @@ describe('The S3 Ingest Granules workflow', () => {
       expect(result).not.toEqual(false);
     });
 
-    it('updates the CMR metadata online resources with the final metadata location', () => {
+    // this test is broken as a result of change in cmrjs package. ignoring for now
+    // until integration tests are back online
+    xit('updates the CMR metadata online resources with the final metadata location', () => {
```
Comment (flamingbear) [inline example/spec/parallel/ingestGranule/IngestGranuleSuccessSpec.js]: This is passing on my stack.

File: packages/integration-tests/api/api.js [hunk 2]
```diff
@@ -121,12 +118,13 @@ async function getAsyncOperation({ prefix, id }) {
...
   const response = await callCumulusApi({
     prefix: prefix,
-    functionName: 'ApiBulkDeleteDefault',
     payload: {
       httpMethod: 'POST',
-      resource: '/bulkDelete',
-      path: '/bulkDelete',
-      pathParameters: {},
+      resource: '/{proxy+}',
+      path: '/bulkDelete/',
```
Comment (flamingbear) [inline packages/integration-tests/api/api.js]: Why did this and `providers` end their paths with a trailing slash?

File: example/spec/parallel/ingestGranule/IngestGranuleSuccessSpec.js (hunk 1 above)
Comment (scisco) [reply example/spec/parallel/ingestGranule/IngestGranuleSuccessSpec.js]: oh great. could you unskip it then?

File: packages/common/key-pair-provider.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ class S3KeyPairProvider {
   static async encrypt(str, keyId = 'public.pub', bucket = null, stack = null) {
     // Download the publickey
     const pki = forge.pki;
-    const b = bucket || process.env.internal;
+    const b = bucket || process.env.internal || process.env.system_bucket;
```
[the remark this answers is not in the document]
Comment (scisco) [reply packages/common/key-pair-provider.js]: sounds good

File: packages/common/key-pair-provider.js [hunk 4]
```diff
@@ -49,7 +49,7 @@ class S3KeyPairProvider {
    */
   static async decrypt(str, keyId = 'private.pem', bucket = null, stack = null) {
     const pki = forge.pki;
-    const b = bucket || process.env.internal;
+    const b = bucket || process.env.internal || process.env.system_bucket;
```
[the remark this answers is not in the document]
Comment (scisco) [reply packages/common/key-pair-provider.js]: agreed

File: packages/integration-tests/api/api.js (hunk 2 above)
Comment (scisco) [reply packages/integration-tests/api/api.js]: either way works

File: packages/api/endpoints/rules.js [hunk 5]
```diff
@@ -108,7 +116,16 @@ async function del(req, res) {
...
   const model = new models.Rule();
 
-  const record = await model.get({ name });
+  let record;
+  try {
+    record = await model.get({ name });
+  }
+  catch (e) {
+    if (e instanceof RecordDoesNotExist) {
+      return res.send({ message: 'No record found' });
```
[the remark this answers is not in the document]
Comment (scisco) [reply packages/api/endpoints/rules.js]: oh, shoot. fixing it now

**Comment by flamingbear** [comment]
One more comment, this branch isn't passing unit testing.  I'm pretty sure I can fix it. 

```
endpoints › endpoints-async-operations › GET /async-operation/{:id} returns a 401 status code if valid authorization is not specified
lerna ERR!
lerna ERR!   /Users/savoie/projects/cumulus/cumulus/packages/api/node_modules/supertest/lib/test.js:268
lerna ERR!
lerna ERR!   Rejected promise returned by test. Reason:
lerna ERR!
lerna ERR!   Error {
lerna ERR!     message: 'expected 401 "Unauthorized", got 404 "Not Found"',
lerna ERR!   }
```
and
```
lerna ERR!   endpoints › endpoints-async-operations › GET /async-operation/{:id} returns the async operation if it does exist
lerna ERR!
lerna ERR!   /Users/savoie/projects/cumulus/cumulus/packages/api/node_modules/supertest/lib/test.js:268
lerna ERR!
lerna ERR!   Rejected promise returned by test. Reason:
lerna ERR!
lerna ERR!   Error {
lerna ERR!     message: 'expected 200 "OK", got 404 "Not Found"',
lerna ERR!   }
lerna ERR!
lerna ERR!   Test._assertStatus (node_modules/supertest/lib/test.js:268:12)
lerna ERR!   Test._assertFunction (node_modules/supertest/lib/test.js:283:11)
lerna ERR!   Test.assert (node_modules/supertest/lib/test.js:173:18)
lerna ERR!   Server.localAssert (node_modules/supertest/lib/test.js:131:12)
```

**Comment by flamingbear** [comment]
but that will be tomorrow

**Pull request merged**